unified_planning/io/pddl_writer.py

```diff
@@ -45,17 +56,17 @@ def convert(self, expression):
 
     def walk_exists(self, expression, args):
         assert len(args) == 1
-        vars_string_list = [f"?{v.name} - {str(v.type)}" for v in expression.variables()]
+        vars_string_list = [f"?{_get_pddl_name(v)} - {self.get_mangled_name(v.type)}" for v in expression.variables()]
```

Comment:
why sometimes you use `_get_pddl_name` and sometimes `self.get_mangled_name`? I think this could cause some name clashes, no?